chips/nrf52/src/spi.rs

```diff
@@ -190,9 +190,22 @@ pub enum Frequency {
     M8 = 0x80000000,
 }
 
-impl From<Frequency> for u32 {
-    fn from(freq: Frequency) -> u32 {
-        match freq {
+impl Frequency {
+    pub fn from_register(reg: u32) -> Option<Frequency> {
+        match reg {
+            0x02000000 => Some(Frequency::K125),
+            0x04000000 => Some(Frequency::K250),
+            0x08000000 => Some(Frequency::K500),
+            0x10000000 => Some(Frequency::M1),
+            0x20000000 => Some(Frequency::M2),
+            0x40000000 => Some(Frequency::M4),
+            0x80000000 => Some(Frequency::M8),
```

Comment:
<img width="682" alt="image" src="https://user-images.githubusercontent.com/1467890/80810083-0d6d3100-8b91-11ea-9380-854233002b7e.png">

Shouldn't this have more fields?

Reply:
Yes, I've stated this in my todos. However the highest frequencies only apply to the nRF52840. I can fix that in this PR too, or create another one for that.